openapi/tag.go

```diff
@@ -17,10 +17,16 @@ type Tag struct {
 
 	// The name of the tag.
 	Name string
+	// A short summary of the tag, used for display purposes.
+	Summary *string
 	// A description for the tag. May contain CommonMark syntax.
 	Description *string
 	// External documentation for this tag.
 	ExternalDocs *oas3.ExternalDocumentation
+	// The name of a tag that this tag is nested under. The named tag must exist in the API description.
+	Parent *string
+	// A machine-readable string to categorize what sort of tag it is.
+	Kind *string
```

Comment:
we should provide along with this some constants for the current registry of known values for this https://spec.openapis.org/registry/tag-kind/